tools/parallelize/parallelize.cpp

```diff
@@ -0,0 +1,203 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "stljobs.h"
+#include <algorithm>
+#include <condition_variable>
+#include <filesystem>
+#include <limits.h>
+#include <memory>
+#include <mutex>
+#include <optional>
+#include <stdexcept>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string>
+#include <string_view>
+#include <thread>
+#include <vector>
+
+#include <Windows.h>
+
+class tp_wait {
+public:
+    tp_wait(PTP_WAIT_CALLBACK pfnwa, PVOID pv, PTP_CALLBACK_ENVIRON pcbe)

```

Comment:
explicit? top-level const?

Reply:
I want this to remain an exact copy of the CreateThreadpoolWait signature.

Reply:
Ok. Do you like explicit on multi-arg ctors?